### ISSUE TITLE
COBOL import: Dimension mistake with certain array definitions

### ISSUE THREAD

**Issue by codemanyak** [fesch/Structorizer.Desktop#1053]
On importing some one-dimensional array definition from COBOL a false type definition (duplicated dimensionality) emerges in the target diagram:
```cobol
      *================================================================ 00072100
      * QUALI-TAB INCLUDES THE QULIFICATION INPUTS UNSORTED.            00072200
      * THESE MUST BE SORTED INTERNAL,AND SORED IN THE SAME TABLE.      00072300
      *================================================================ 00072400
       01 QUALI-TAB.                                                    00072500
           05 QUALIES  OCCURS 5    PIC 9(1).                            00072600
      *================================================================ 00072700
```
describes a simple on-dimensional array of int but the puzzling result in Structorizer looks as follows:
![grafik](https://user-images.githubusercontent.com/15326471/182004017-132f53c8-2197-49b0-9976-a5dd7672141b.png)

**Comment by codemanyak** [comment]
Possibly the defect is induced by the order of PIC vs. OCCURS - in a slightly differing example the resulting type is correct:
```cobol
*> Demonstration descriptions. 
    01  tabl-demos. 
        02 tabl-demo                      pic x(40)
            occurs 23 times. 
    01  max-demo                       pic 9(04) binary value 23. 
    01  sub-demo                       pic 9(04) binary. 
```
leads to a correct type definition:
![grafik](https://user-images.githubusercontent.com/15326471/182004208-5a11ab70-c8b3-44b7-a762-f77e84383b62.png)

**Comment by codemanyak** [comment]
It turned out that it has nothing to do with the order of the data descriptions. Instead this bug is the consequence of an incompetent fix for #695.
The original problem was that `pic x(n)` data and data without usage declaration were truly regarded as strings but then additional `occurs` declarations were ignored for strings in method `getTypeString(CobVar, boolean)`, such that the numeric and string types behaved differently. In order to fix it (in the wrong place) for strings the result became wrong for numeric arrays.
